sdk/eventgrid/azure-messaging-eventgrid/src/test/java/com/azure/messaging/eventgrid/EventGridPublisherClientTests.java

```diff
@@ -234,6 +246,71 @@ public void publishCloudEvent() {
+                .addPolicy((httpPipelineCallContext, httpPipelineNextPolicy) -> {
+                    HttpHeader httpHeader = httpPipelineCallContext.getHttpRequest().getHeaders().get("aeg-channel-name");
+                    assertNotNull(httpHeader);
+                    return httpPipelineNextPolicy.process();
+                })
+                .buildCloudEventPublisherAsyncClient();
+
+        CloudEvent event = new CloudEvent("/microsoft/testEvent", "Microsoft.MockPublisher.TestEvent",
+                BinaryData.fromObject(new HashMap<String, String>() {
+                    {
+                        put("Field1", "Value1");
+                        put("Field2", "Value2");
+                        put("Field3", "Value3");
+                    }
+                }), CloudEventDataFormat.JSON, "application/json")
+                .setSubject("Test")
+                .setTime(OffsetDateTime.now());
+
+        Mono<Response<Void>> responseMono = egClient.sendEventsWithResponse(Arrays.asList(event), new SendEventsOptions()
+                .setChannelName(getChannelName(EVENTGRID_PARTNER_CHANNEL_NAME)));
+        StepVerifier.create(responseMono)
+                .assertNext(response -> assertEquals(200, response.getStatusCode()))
+                .verifyComplete();
+    }
+
+    @Test
+    public void publishEventGridEventToPartnerTopic() {
+        EventGridPublisherAsyncClient<EventGridEvent> egClient = builder
+                .endpoint(getEndpoint(EVENTGRID_PARTNER_NAMESPACE_TOPIC_ENDPOINT))
+                .credential(getKey(EVENTGRID_PARTNER_NAMESPACE_TOPIC_KEY))
+                .addPolicy((httpPipelineCallContext, httpPipelineNextPolicy) -> {
+                    HttpHeader httpHeader = httpPipelineCallContext.getHttpRequest().getHeaders().get("aeg-channel-name");
+                    assertNotNull(httpHeader);
+                    return httpPipelineNextPolicy.process();
+                })
+                .buildEventGridEventPublisherAsyncClient();
+
+        EventGridEvent event = new EventGridEvent("Test", "Microsoft.MockPublisher.TestEvent",
+                BinaryData.fromObject(new HashMap<String, String>() {
+                    {
+                        put("Field1", "Value1");
+                        put("Field2", "Value2");
+                        put("Field3", "Value3");
+                    }
+                }),
+                "1.0")
+                .setEventTime(OffsetDateTime.now());
+
+        Mono<Response<Void>> responseMono = egClient.sendEventsWithResponse(Arrays.asList(event),
+                new SendEventsOptions().setChannelName(getChannelName(EVENTGRID_PARTNER_CHANNEL_NAME)));
+        StepVerifier.create(responseMono)
+                .expectErrorSatisfies(exception -> {
+                    assertEquals(HttpResponseException.class.getName(), exception.getClass().getName());
+                    if (exception instanceof HttpResponseException) {
```

Comment:
since we already assertEquats the class name, this instanceof  statement checking is redundant

Reply:
This will cause checkstyle error if we directly cast without checking for `instanceof`.